File: ios/MapsHereConfig.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MapsHereConfig, NSObject)


//RCT_EXTERN_METHOD(initializeHereSDK:(NSString *)accessKeyID
//                  withAccessKeySecret:(NSString *)accessKeySecret
//                  withResolver:(RCTPromiseResolveBlock)resolve
//                  withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN__BLOCKING_SYNCHRONOUS_METHOD(initializeHereSDK:(NSString *)accessKeyID
                                        withAccessKeySecret:(NSString *)accessKeySecret)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/MapsHereConfig.swift
import heresdk

@objc(MapsHereConfig)
class MapsHereConfig: NSObject {
    
    @objc(initializeHereSDK:withAccessKeySecret:)
    func initializeHereSDK(accessKeyID: String, accessKeySecret: String) -> String {
            let options = SDKOptions(
                accessKeyId: accessKeyID,
                accessKeySecret: accessKeySecret)
            
            do {
                try SDKNativeEngine.makeSharedInstance(options: options)
                return "SDKNativeEngine started"
//                resolve("Here SDK initialized")
            } catch _ {
                return "SDKNativeEngine errored"
//                reject("", "", error)
            }
        }
}

